src/runtime.cc

```diff
@@ -215,23 +217,8 @@ namespace detail {
 	}
 
 	void runtime::sync() noexcept {
-		// (Note: currently this function busy waits, but sync is slow and shouldn't be on the critical path anyway)
-		sync_id++;
-
-		// First, broadcast SYNC command once the scheduler has finished all previous tasks
-		if(is_master_node()) {
-			while(!schdlr->is_idle()) {
-				std::this_thread::yield();
-			}
-			sync_data cmd_data{};
-			cmd_data.sync_id = sync_id;
-			broadcast_control_command(command_type::SYNC, cmd_data);
-		}
-
-		// Then we wait for that sync to actually be reached.
-		while(exec->get_highest_executed_sync_id() < sync_id) {
-			std::this_thread::yield();
-		}
+		const auto epoch = task_mngr->finish_epoch(epoch_action::barrier);
```

Comment:
In this context I feel like `finish_epoch` is not a great name. Here, we basically start waiting for an epoch after "finishing" it. Maybe `generate_epoch`?

Reply:
While we're at it, we should improve the naming of `create_task` as well. It does not fit with the other two because it can actually generate more than one task (if a task horizon is due).

I suggest

1. `submit` or `submit_command_group` for create_task (because it is exactly `distr_queue::submit`)
3. `generate_epoch_task` instead of `finish_epoch`
2. `generate_horizon_task` instead of `generate_task_horizon` (for consistency)

Reply:
Done.